doc/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,143 @@
+
+```
+$ dd if=/dev/random of=out seek=1 count=1
+dd: writing to 'out': Invalid argument
+```
+
+In Mountpoint's logs, a warning message will be emitted similar to below:
+
+```
+WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: 
+mountpoint_s3::fuse: write failed:upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
+```
+
+## Writing to an existing File 
+
+Mountpoint support overwriting to existing file using mount option `--allow-overwrite` . 
+Trying to open an existing file for writing without `--allow-overwrite`  flag will fail with the error: `Operation not permitted`.
+For example, there is an pre-existing file 'existing-file.txt' in mounted directory. We try to overwrite the 'existing-file' as follow:
+
+```
+$ echo "Overwriting a file..." > existing-file.txt
+operation not permitted: existing-file.txt
+```
+
+Log entries for file overwriting looks like:
+
+```
+WARN setattr{req=4 ino=21 name="existing-file.txt"}: 
+mountpoint_s3::fuse: setattr failed: inode error: inode 21 (full key "existing-file.txt") is a remote inode and its attributes cannot be modified
+```
+
+### Unreleased
+
+With improvement in error logging, we will get the following logs for file overwrite-
+
+In the logs, we get the following WARN message:
+
+```
+WARN setattr{req=11 ino=2 name="existing-file.txt"}: 
+mountpoint_s3::fuse: setattr failed: file overwrite is disabled by default, you need to remount with --allow-overwrite flag and open the file in truncate mode (O_TRUNC) to overwrite it
+```
+
+## Deleting file
+
+In order to delete files using Mountpoint user need to enable it using `--allow-delete` CLI flag. To know more about the behaviour of file deletion, please visit [Delete Semantics](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#deletes)
+Without the flag, if a customer tries to delete, for example test-file.txt, they will get the following error:
+
+```
+$ rm test-file.txt
+rm: cannot remove 'test-file.txt': Operation not permitted
+```
+
+In Mountpoint's logs, following message will be emitted:
+
+```
+WARN unlink{req=8 parent=1 name="test-file.txt"}: 
+mountpoint_s3::fuse: unlink failed: Deletes are disabled. Use '--allow-delete' mount option to enable it.
+```
+
+## Listing file which is shadowed by directory of same name
+
+A file system directory cannot contain both a file and a directory of the same name. If your bucket's directory structure would result in this state, only the directory will be accessible. The object will not be accessible. 
+For more details on how Mountpoint maps S3 object keys to files and directories, see the [semantics documentation](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#mapping-s3-object-keys-to-files-and-directories). 
+For example, our bucket contains the following object keys:
+
+* out
+* out/image.jpg
```

Comment:
Unless it makes sense (like `out` file), let's use things like `a`/`b` rather than `out`.